### ISSUE TITLE
Release workflow

### ISSUE THREAD

**Issue by jenniferplusplus** [Letterbook/Letterbook.ActivityPub#14]
The current publishing workflow only produces prelease packages, tagged like `-ci0000-whatever`. We also want a release version publishing workflow. In an ideal world, it would have auto-generated release notes, but also not require semantic commits because those are incredibly tedious.

**Comment by runewake2** [comment]
So one issue with how things currently work is that Nuget works best with semantic versioning so you could do something like 

```xml
<PackageReference Include="Letterbook.ActivityPub" Version="1.0.*" />
```

In your `.csproj` file to support all 1.0 releases. The extra bits break that support right now.

Are you picturing a release every push to main, when main is tagged or something else? Happy to take a look at some of this if you'd like.

IDK about magically generating release notes without some format unless we try to have some sort of synopsis generation. Maybe a list of closed issues contained in that release?

**Comment by jenniferplusplus** [comment]
Yes, I'm imagining publishing a prerelease build on every merge to main, and then we can publish release versions by tagging them.

Right now ci uses gittools to generate the next prerelease version number. The value in the csproj isn't actually used for anything.

https://github.com/Letterbook/Letterbook.ActivityPub/blob/a62fa28e0a2146ad1cffdb2df1f8ee658d733643/.github/workflows/pull_requests.yml#L43
